### ISSUE TITLE
bug in latex drawer barrier

### ISSUE THREAD

**Issue by ajavadia** [Qiskit/qiskit-terra#1124]
### Informations

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?
When drawing this circuit, the barrier on the last qubit is not respected. 
```python
from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
from qiskit.tools.visualization import circuit_drawer
q = QuantumRegister(3, 'q')
c = ClassicalRegister(3, 'c')
circ = QuantumCircuit(q, c)
circ.x(q[0])
circ.x(q[1])
circ.barrier(q)
circ.measure(q, c)
circuit_drawer(circ, output='latex')
```

![image](https://user-images.githubusercontent.com/8622381/47267192-53fabd80-d50e-11e8-9325-7fb46a6eac27.png)

The `mpl` drawer works fine.

**Comment by mtreinish** [comment]
Hmm, this one is gonna be a bit trickier to fix. The issue is the barrier isn't actually an element in the underlying xymatrix that qcircuit is building. So to fix this we'll have to iterate over the rows and somehow figure out that we need to insert a \qw at that column and move the \measure over. I'm not exactly sure how to do that though.

**Comment by jaygambetta** [comment]
is there an update on this. 
Is this for 0.7

**Comment by charlesastaylor** [comment]
Assuming there isn't an update, I could try and have a look at this. It seems like it is quite a bit more in depth than the other issues I've had a go at so far so not sure how long it will take me or whether I'll be even be able to do it! But it noone else is trying I'm happy to give it a go!

**Comment by mtreinish** [comment]
I have not had a chance to work on this (I moved it to in progress in the project by mistake) so there is no update on this. I wasn't going to wait on this for 0.7.

@charlesastaylor if you want to give it a try, go for it. I can give you help if you need it. I have a feeling it'll be annoying to fix though since we'll have to at least iterate over the circuit one more time at the end to figure out whether we'll need to space things out.

**Comment by mtreinish** [comment]
Just leaving a comment here that #1306 was a duplicate of this issue, but manifested itself differently by raising a `VisualizationError` with the error message: `Error during Latex building: invalid literal for int() with base 10: '1} \\\\barrier{1'` you can look at the closing for duplicate comment: https://github.com/Qiskit/qiskit-terra/issues/1306#issuecomment-441735166 for more details on why/how.
